platform/darwin/src/MGLBackgroundStyleLayer.mm

```diff
@@ -35,6 +35,10 @@ - (instancetype)initWithIdentifier:(NSString *)identifier
 
 - (void)addToMapView:(MGLMapView *)mapView
 {
+    if (_pendingLayer == nullptr) {
+        [NSException raise:NSInternalInconsistencyException format:@"Adding the same instance to the style more than once is invalid."];
```

Comment:
`NSInternalInconsistencyException` is the exception name used for assertions. Either define a custom exception name like `MGLRedundantLayerException` or use `NSGenericException`. #7258 isn’t critical, so it’s fine to specify `MGLRedundantLayerException` as a string literal for now without defining a constant.